test/test_srv_polling.py

```diff
@@ -230,6 +230,50 @@ def erroring_seedlist():
             raise ConfigurationError
         self._test_recover_from_initial(erroring_seedlist)
 
+    def test_10_all_dns_selected(self):
+        response = [("localhost.test.build.10gen.cc", 27017),
+                    ("localhost.test.build.10gen.cc", 27017),
+                    ("localhost.test.build.10gen.cc", 27019),
+                    ("localhost.test.build.10gen.cc", 27020)]
+
+        def nodelist_callback():
+            return response
+        with SrvPollingKnobs(ttl_time=WAIT_TIME,
+                             min_srv_rescan_interval=WAIT_TIME):
+            client = MongoClient(self.CONNECTION_STRING, srvMaxHosts=0)
+            with SrvPollingKnobs(nodelist_callback=nodelist_callback):
+                self.assert_nodelist_change(response, client)
+                sleep(2*common.MIN_SRV_RESCAN_INTERVAL)
+                expected_topology = {
+                    ("localhost.test.build.10gen.cc", 27017),
+                    ("localhost.test.build.10gen.cc",  27019),
+                    ("localhost.test.build.10gen.cc",  27020)}
+                final_topology = set(
+                    client.topology_description.server_descriptions())
+                self.assertSetEqual(final_topology, final_topology |
+                                  expected_topology)
+
+    def test_11_all_dns_selected(self):
+        response = [("localhost.test.build.10gen.cc", 27019),
+                    ("localhost.test.build.10gen.cc", 27020)]
+
+        def nodelist_callback():
+            return response
+
+        with SrvPollingKnobs(
+                ttl_time=WAIT_TIME, min_srv_rescan_interval=WAIT_TIME):
+            client = MongoClient(self.CONNECTION_STRING, srvMaxHosts=2)
+            with SrvPollingKnobs(nodelist_callback=nodelist_callback):
+                self.assert_nodelist_change(response, client)
```

Comment:
This is the assertion that fails:
```
 [2021/10/21 19:40:48.759] FAIL [10.041s]: test_11_all_dns_selected (test_srv_polling.TestSrvPolling)
 [2021/10/21 19:40:48.759] ----------------------------------------------------------------------
 [2021/10/21 19:40:48.760] Traceback (most recent call last):
 [2021/10/21 19:40:48.760]   File "/data/mci/6c8659548e53884acc9814e9e91b6644/src/test/test_srv_polling.py", line 267, in test_11_all_dns_selected
 [2021/10/21 19:40:48.760]     self.assert_nodelist_change(response, client)
 [2021/10/21 19:40:48.760]   File "/data/mci/6c8659548e53884acc9814e9e91b6644/src/test/test_srv_polling.py", line 116, in assert_nodelist_change
 [2021/10/21 19:40:48.760]     wait_until(predicate, "see expected nodelist", timeout=100*WAIT_TIME)
 [2021/10/21 19:40:48.760]   File "/data/mci/6c8659548e53884acc9814e9e91b6644/src/test/utils.py", line 755, in wait_until
 [2021/10/21 19:40:48.760]     raise AssertionError("Didn't ever %s" % success_description)
 [2021/10/21 19:40:48.760] AssertionError: Didn't ever see expected nodelist
```
What is the current TopologyDescription when it fails? That would help determine what is not working as expected.

Reply:
<TopologyDescription id: 6171c8203509121876af7cf4, topology_type: Unknown, servers: [<ServerDescription ('localhost.test.build.10gen.cc', 27017) server_type: Unknown, rtt: None>, <ServerDescription ('localhost.test.build.10gen.cc', 27018) server_type: Unknown, rtt: None>]>

Reply:
Hmm, that's unexpected. Try debugging _updated_topology_description_srv_polling. In this test it's supposed to be called with nodes [27019, 27020]. That should cause 27017 and 27018 to be removed and replaced with 27019 and 27020.